### PR TITLE
Dashboard/feature/no internet connection

### PR DESCRIPTION
**Pull request by hassanwasfy**
- [x] handle no internet connection
- [x] refactor tryToExcute function in the baseGateWay

### CODE REVIEW

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/BaseGateway.kt [hunk 1]
```diff
@@ -19,14 +21,15 @@ abstract class BaseGateway {
...
             val errorMessages = e.response.body<ServerResponse<String>>().status?.errorMessages
             errorMessages?.let(::throwMatchingException)
-            throw UnknownErrorException(e.message)
-        } catch (e: ConnectException) {
-            throw NoInternetException()
-        }catch(e: ConnectException){
-            throw NoInternetException()
         } catch (e: Exception) {
-            throw UnknownErrorException(e.message.toString())
+            when (e) {
```
Comment (Asiasama710) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/BaseGateway.kt:25]: u don't need use `when ` using more one catch directly

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/main/MainScreenModel.kt [hunk 2]
```diff
@@ -35,13 +35,21 @@ class MainScreenModel(
...
+                hasInternetConnection = true
             )
         }
     }
 
     private fun onError(error: ErrorState) {
-        updateState { it.copy(error = error) }
+        when(error){
+            is ErrorState.NoConnection -> {
+                updateState { it.copy(error = error, hasInternetConnection = true) }
```
Comment (Asiasama710) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/main/MainScreenModel.kt:47]: u can remove `error ` i think we don't need it

Comment (hassanwasfy) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/main/MainScreenModel.kt:47]: if not specified will not mapp the right error, I made it seperate.

**Review by AzizaHelmy — APPROVED** [review]
good refactor 👏

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/table/BpTable.kt [hunk 3]
```diff
@@ -31,6 +31,7 @@ data class Header(val text: String, val weight: Float = 1f)
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun <T> ColumnScope.BpTable(
+    hasConnection: Boolean = true,
```
Comment (ahmednasserzaza) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/table/BpTable.kt:34]: just the arrangement of parameters , parameters that hasn't default value recommended to be first

**Review by ahmednasserzaza — APPROVED** [review]
Super Excellent 👏❤

**Review by HaidyAbuGom3a — APPROVED** [review]

**Review by Asiasama710 — APPROVED** [review]

**Pull request merged**